prow/git/client.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+import (
+	"os"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Clients knows how to create clients for repos
+type Clients interface {
+	Repo(org, repo, dir string) (RepoClient, error)
+}
+
+// RepoClient exposes interactions with a git repo
+type RepoClient interface {
+	Publisher
+}
+
+func NewClients(host string, useSSH bool, user func() (name, email, login string, err error), token func() []byte, censor func(content []byte) []byte) Clients {
```

Comment:
Nit: why do we return an interface instead of exporting the struct? Interfaces should IMHO be defined on the receiving side

Reply:
If we allow someone to receive a struct, they will, and then we're screwed if we want to change something behind the scenes